### ISSUE TITLE
vue/require-valid-default-prop fails to report error on type mismatch

### ISSUE THREAD

**Issue by jantajov** [vuejs/eslint-plugin-vue#1044]
Given following single file Vue.js component:

```vue
<template>
  <div>Hello</div>
</template>

<script>
  export default {
    name: 'AppHello',
    props: {
      textProp: {type: String, default: () => 1233}
    }
  }
</script>
```

I would expect to get `vue/require-valid-default-prop` error as the prop does not have valid default value. Types does not match.

Howsoever, no such error is thrown during linting.

I am not sure if this is intended. But if it is, at least the [docs](https://eslint.vuejs.org/rules/require-valid-default-prop.html#vue-require-valid-default-prop) should be updated.

Versions: 
```
vue/cli-plugin-eslint is ^3.11.0"
eslint is ^5.16.0
node is ^10.16.0
```

**Comment by ota-meshi** [comment]
Thank you for this issue.

Now, the `vue/require-valid-default-prop` rule does not check what the function returns.
https://github.com/vuejs/eslint-plugin-vue/pull/119#discussion_r131395839

I think we need to add a check to `vue/require-valid-default-prop` rule.
